### PR TITLE
Fixed problems with tab completion after moving tabs (#2141)

### PR DESCRIPTION
**Pull request by Raihid**
It was a bit hard to recreate the issue #2141 in tests, since the completer seems to update the pages by itself from time to time. The current feature test is a bit weird, but indeed fails in the old version and passes in the new one.
Not really sure if I changed `TabbedBrowserStub` test helper correctly (I had to, since in the old version it didn't have tabBar() method).

### CODE REVIEW

File: qutebrowser/browser/commands.py [hunk 1]
```diff
@@ -1028,17 +1028,13 @@ def tab_move(self, index: typing.Union[str, int]=None, count=None):
             raise cmdexc.CommandError("Can't move tab to position {}!".format(
                 new_idx + 1))
 
-        tab = self._current_widget()
         cur_idx = self._current_index()
-        icon = self._tabbed_browser.tabIcon(cur_idx)
-        label = self._tabbed_browser.page_title(cur_idx)
         cmdutils.check_overflow(cur_idx, 'int')
         cmdutils.check_overflow(new_idx, 'int')
         self._tabbed_browser.setUpdatesEnabled(False)
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: You can most likely remove that (and the try/finally) too, as we shouldn't have any flickering from removing/inserting tabs anymore now.

File: tests/helpers/stubs.py [hunk 2]
```diff
@@ -584,6 +585,9 @@ def page_title(self, i):
     def on_tab_close_requested(self, idx):
         del self.tabs[idx]
 
+    def tabBar(self):
+        return QTabBar(self._qtabbar)
```
Comment (The-Compiler) [inline tests/helpers/stubs.py]: Why call the `QTabBar` constructor again here (which probably just copies the existing one)?

Comment (Raihid) [inline tests/helpers/stubs.py]: I've simply overlooked this when adding `_qtabbar` as a variable 🙂 Thanks!

File: qutebrowser/browser/commands.py [hunk 3]
```diff
@@ -1028,21 +1028,13 @@ def tab_move(self, index: typing.Union[str, int]=None, count=None):
...
-        self._tabbed_browser.setUpdatesEnabled(False)
-        try:
-            color = self._tabbed_browser.tab_indicator_color(cur_idx)
-            self._tabbed_browser.removeTab(cur_idx)
-            self._tabbed_browser.insertTab(new_idx, tab, icon, label)
-            self._set_current_index(new_idx)
-            self._tabbed_browser.set_tab_indicator_color(new_idx, color)
-        finally:
-            self._tabbed_browser.setUpdatesEnabled(True)
+        color = self._tabbed_browser.tab_indicator_color(cur_idx)
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: I didn't realize this before, but I think we don't need the indicator color saving/restoring anymore, I'd expect Qt to handle that - can you remove it and verify moving a tab doesn't reset its indicator color?

File: qutebrowser/browser/commands.py [hunk 4]
```diff
@@ -1028,21 +1028,13 @@ def tab_move(self, index: typing.Union[str, int]=None, count=None):
...
+        self._tabbed_browser.tabBar().moveTab(cur_idx, new_idx)
+        self._set_current_index(new_idx)
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: Also not sure if we still need that - does the focus persist on the now-moved tab when we don't do this?

**Comment by The-Compiler** [comment]
Sure, no hurry!

**Comment by Raihid** [comment]
Yup, everything looks alright without those lines :+1:

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**